Plugins/nosUtilities/Source/Ring.h

```diff
@@ -1009,44 +1017,44 @@ struct RingNodeBase : NodeContext
 		if (OnRestart == OnRestartType::WAIT_UNTIL_FULL)
 			Mode = RingMode::FILL;
 		if (Ring)
-		{
 			Ring->Stop();
-		}
 	}
 
 	void OnPathStart() override
 	{
 		if (!Ring) return;
 		// Reset read pool for Queues(OnRestart::RESET) and Rings(OnRestart::WAIT_UNTIL_FULL) with Repeat too, if repeat, then size-1 CopyFroms will be repeated from the last buffer.
 		if (OnRestart == OnRestartType::RESET || RepeatWhenFilling)
-		{
 			Ring->Reset(false);
-		}
 		// We must wait for at least a frame to be sure that providing path is started and running smoothly
-		else if (Ring->IsFull())
+		else if (Ring->IsFull() && !Ring->Read.Pool.empty())
 		{
 			Ring->Write.Pool.push_back(Ring->Read.Pool.front());
 			Ring->Read.Pool.pop_front();
 		}
 		if (RequestedRingSize)
 		{
 			Ring->Resize(*RequestedRingSize);
-			RequestedRingSize = std::nullopt;
```

Comment:
Are we sure about this? @caner-milko 
RequestedRingSize doesn't seem to be set to nullopt, which may cause Ring->Resize() to be called frequently